alvr/server/src/connection.rs

```diff
@@ -748,19 +778,69 @@ fn try_connect(mut client_ips: HashMap<IpAddr, String>) -> ConResult {
                 if let Some(stats) = &mut *STATISTICS_MANAGER.lock() {
                     stats.report_tracking_received(tracking.target_timestamp);
 
+                    // Handle hand gestures
+                    if let Switch::Enabled(controllers_config) = &settings.headset.controllers {
+                        if let Switch::Enabled(gestures_config) =
+                            &controllers_config.hand_tracking.use_gestures
+                        {
+                            let mut hand_gesture_manager_lock = hand_gesture_manager.lock();
+
+                            if tracking.hand_skeletons[0].is_some() {
+                                trigger_hand_gesture_actions(
+                                    gestures_button_mapping_manager.as_mut().unwrap(),
+                                    *LEFT_HAND_ID,
+                                    &hand_gesture_manager_lock.get_active_gestures(
+                                        tracking.hand_skeletons[0].unwrap(),
+                                        gestures_config.clone(),
+                                        *LEFT_HAND_ID,
+                                    ),
+                                );
+                            }
+                            if tracking.hand_skeletons[1].is_some() {
+                                trigger_hand_gesture_actions(
+                                    gestures_button_mapping_manager.as_mut().unwrap(),
+                                    *RIGHT_HAND_ID,
+                                    &hand_gesture_manager_lock.get_active_gestures(
+                                        tracking.hand_skeletons[1].unwrap(),
+                                        gestures_config.clone(),
+                                        *RIGHT_HAND_ID,
+                                    ),
+                                );
+                            }
+                        }
+                    }
+
+                    let data_manager_lock = SERVER_DATA_MANAGER.read();
+                    let config = &data_manager_lock.settings().headset;
+
+                    let hand_skeletons_enabled =
+                        if let Switch::Enabled(controllers) = &config.controllers {
+                            controllers.hand_tracking.enable_skeleton
+                        } else {
+                            false
+                        };
```

Comment:
Instead of creating a `hand_skeletons_enabled` variable, I think you can conditionally set `ffi_left_hand_skeleton` and `ffi_right_hand_skeleton` to None.